src/lug/Graphics/Vulkan/Gui.cpp

```diff
@@ -0,0 +1,876 @@
+        break;
+    case (lug::Window::Event::Type::CharEntered):
+        if (event.character.val > 0 && event.character.val < 0x10000) {
+            io.AddInputCharacter(static_cast<unsigned short>(event.character.val));
+        }
+        break;
+    default:
+        break;
+    }
+}
+
+const Vulkan::API::Semaphore& Gui::getGuiSemaphore(uint32_t currentImageIndex) const {
+    return _framesData[currentImageIndex].semaphore;
+}
+
+void Gui::updateBuffers(uint32_t currentImageIndex) {
+    ImDrawData* imDrawData = ImGui::GetDrawData();
+    if (!imDrawData) {
+        return;
+    }
+    API::Device &device = _renderer.getDevice();
+    FrameData& frameData = _framesData[currentImageIndex];
+
+    // Note: Alignment is done inside buffer creation
+    VkDeviceSize vertexBufferSize = imDrawData->TotalVtxCount * sizeof(ImDrawVert);
+    VkDeviceSize indexBufferSize = imDrawData->TotalIdxCount * sizeof(ImDrawIdx);
+
+    // Update buffers only if vertex or index count has been changed compared to current buffer size
+    {
+        // Vertex buffer
+        if ((static_cast<VkBuffer>(frameData.vertexBuffer) == VK_NULL_HANDLE) || (static_cast<int>(frameData.vertexBuffer.getRequirements().size) != imDrawData->TotalVtxCount)) {
+            {
+                // Create Vertex buffer
+                {
+                    API::Builder::Buffer bufferBuilder(device);
+                    bufferBuilder.setQueueFamilyIndices({ _transferQueue->getQueueFamily()->getIdx() });
+                    bufferBuilder.setSize(vertexBufferSize);
+                    bufferBuilder.setUsage(VK_BUFFER_USAGE_VERTEX_BUFFER_BIT);
+
+                    VkResult result{VK_SUCCESS};
+                    if (!bufferBuilder.build(frameData.vertexBuffer, &result)) {
+                        LUG_LOG.error("Gui::initFontsTexture: Can't create staging buffer: {}", result);
+                        return ;
+                    }
+                }
+
+                if (frameData.vertexMemoryPtr) {
+                    frameData.vertexMemory.unmap();
+                }
+
+                API::Builder::DeviceMemory deviceMemoryBuilder(device);
+                deviceMemoryBuilder.setMemoryFlags(VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT);
+                deviceMemoryBuilder.addBuffer(frameData.vertexBuffer);
+
+                VkResult result{VK_SUCCESS};
+                if (!deviceMemoryBuilder.build(frameData.vertexMemory, &result)) {
+                    LUG_LOG.error("Gui::initFontsTexture: Can't create staging buffer device memory: {}", result);
+                    return ;
+                }
+
+                frameData.vertexMemoryPtr = frameData.vertexMemory.mapBuffer(frameData.vertexBuffer);
+            }
+        }
+
+        // IndexBuffer
+        if ((static_cast<VkBuffer>(frameData.indexBuffer) == VK_NULL_HANDLE) || (static_cast<int>(frameData.indexBuffer.getRequirements().size) < imDrawData->TotalIdxCount)) {
```

Comment:
You are comparing a number of indices and a size of buffer!
And if the size required is lesser, re-use the buffer.